### PR TITLE
feat(*): add calculation for total token balance per shard

### PR DESCRIPTION
**Pull request by DZariusz**
This feature is required by andromeda.
We need total token balance per shard, so we can calculate total stake for value.
This will allow us to decide, if election was valid or not.
Also fix some linters errors.

LIT-93

### CODE REVIEW

File: contracts/VerifierRegistry.sol [hunk 1]
```diff
@@ -95,28 +106,34 @@ contract VerifierRegistry is Ownable {
...
-    require(_value > 0 && verifier.balance >= _value);
+    require(_value > 0 && verifier.balance >= _value, "nothing to withdraw");
 
     verifier.balance -= _value;
 
+    uint256 shard = verifier.shard;
+    uint256 shardBalance = totalBalancePerShard[shard] - _value;
+    totalBalancePerShard[shard] = shardBalance;
+
+    emit LogTotalBalancePerShard(shard, shardBalance);
```
Comment (therevoltingx) [inline contracts/VerifierRegistry.sol]: This should reflect the action/event that just happened. 
Something like: `LogVerifierWithdrawal` or `LogVerifierBalanceUpdate`

File: contracts/VerifierRegistry.sol [hunk 2]
```diff
@@ -31,19 +33,22 @@ contract VerifierRegistry is Ownable {
 
   mapping(address => Verifier) public verifiers;
 
+  mapping(uint256 => uint256) public totalBalancePerShard;
```
Comment (therevoltingx) [inline contracts/VerifierRegistry.sol]: `totalBalancePerShard` seems a bit redundant, also `balance` should be plural.

How about just: `balances`.

File: contracts/VerifierRegistry.sol (hunk 1 above)
Comment (DZariusz) [reply contracts/VerifierRegistry.sol]: Event for transfer/withdraw is present in `Token` contract.`LogTotalBalancePerShard` event is only about shardBalance.

File: contracts/VerifierRegistry.sol (hunk 2 above)
Comment (DZariusz) [reply contracts/VerifierRegistry.sol]: is `balancesPerShard` would be ok? `balances` are usually uses as mapping for addresses, also we have `veriefiersPerShard` variable, so I follow this convention.

**Comment by therevoltingx** [comment]
👍

**Comment by enddynayn** [comment]
👍

**Pull request merged**